tokio-test/src/macros.rs

```diff
@@ -0,0 +1,109 @@
+//! A collection of useful macros for testing futures and tokio based code
+
+/// Assert if a poll is ready
+#[macro_export]
+macro_rules! assert_ready {
+    ($e:expr) => {{
+        match $e {
+            Ok(::futures::Async::Ready(v)) => v,
+            Ok(_) => panic!("not ready"),
+            Err(e) => panic!("error = {:?}", e),
+        }
+    }};
+    ($e:expr, $($msg:expr),+) => {{
+        match $e {
+            Ok(::futures::Async::Ready(v)) => v,
+            Ok(_) => {
+                let msg = format!($($msg),+);
+                panic!("not ready; {}", msg)
+            }
+            Err(e) => {
+                let msg = format!($($msg),+);
+                panic!("error = {:?}; {}", e, msg)
+            }
+        }
+    }};
+}
+
+/// Asset if the poll is not ready
+#[macro_export]
+macro_rules! assert_not_ready {
+    ($e:expr) => {{
+        match $e {
+            Ok(::futures::Async::NotReady) => {}
+            Ok(::futures::Async::Ready(v)) => panic!("ready; value = {:?}", v),
+            Err(e) => panic!("error = {:?}", e),
+        }
+    }};
+    ($e:expr, $($msg:expr),+) => {{
+        match $e {
+            Ok(::futures::Async::NotReady) => {}
+            Ok(::futures::Async::Ready(v)) => {
+                let msg = format!($($msg),+);
+                panic!("ready; value = {:?}; {}", v, msg)
+            }
+            Err(e) => {
+                let msg = format!($($msg),+);
+                panic!("error = {:?}; {}", e, msg)
+            }
+        }
+    }};
+}
+
+/// Assert if a poll is ready and check for equality on the value
+#[macro_export]
+macro_rules! assert_ready_eq {
+    ($e:expr, $expect:expr) => {
+        assert_eq!($e, ::futures::Async::Ready($expect));
+    };
+
+    ($e:expr, $expect:expr, $($msg:expr),+) => {
+        assert_eq!($e, ::futures::Async::Ready($expect), $msg);
```

Comment:
You need to actually repeat out the message here:
```suggestion
        assert_eq!($e, ::futures::Async::Ready($expect), $($msg)+);
```
(i'm surprised this compiles?)

Reply:
> (i'm surprised this compiles?)

_me_irl_

Reply:
Ah, I see: it only errors if you actually _use_ the macro: [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=8295a5157477207eace26897b498e9d6) compiles, but [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=592f96550d12f37d633378ebbabe5ed6) doesn't.